packages/ref-core/src/ref_core/metrics.py

```diff
@@ -108,18 +165,14 @@ class Metric(Protocol):
     but multiple providers can implement the same metric.
     """
 
-    # input_variable: list[VariableDefinition]
+    inputs: list[DataRequirement]
```

Comment:
Should this be a list of requirements or split them out?
```suggestion
    source_requirements: DataRequirement | None
    reference_requirements: DataRequirement | None
    metric_requirements: DataRequirement | None
```

As a non-expert the data sources are interchangeable as a change in any dependent data would trigger a calculation, but as a metric developer there might be other opinions. IMHO how the trigger information is presented to the metric calculation is somewhat independent to this.

Pros for single field: No schema changes if additional dimensions of datasets are added, supports the concept of multiple input sources (CMIP6 + CMIP7).
Cons: requires knowing to add additional dependencies on different data types rather than it being more explicit

Pros for multiple: More explicit, don't need to deal with the potential of multiple duplicate sources.
Cons: Less flexible